server/datastore/mysql/hosts_test.go

```diff
@@ -3133,6 +3157,120 @@ func testHostsNoSeenTime(t *testing.T, ds *Datastore) {
 	assert.Equal(t, uint(0), metrics.MissingInActionHosts)
 }
 
+func testHostsListHostDeviceMapping(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	h, err := ds.NewHost(ctx, &fleet.Host{
+		ID:              1,
+		OsqueryHostID:   "1",
+		NodeKey:         "1",
+		Platform:        "linux",
+		Hostname:        "host1",
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		PolicyUpdatedAt: time.Now(),
+		SeenTime:        time.Now(),
+	})
+	require.NoError(t, err)
+
+	ds.writer.ExecContext(ctx, `INSERT INTO host_emails (host_id, email, source) VALUES (?, ?, ?)`,
+		h.ID, "a@b.c", "src1")
+	ds.writer.ExecContext(ctx, `INSERT INTO host_emails (host_id, email, source) VALUES (?, ?, ?)`,
+		h.ID, "b@b.c", "src1")
```

Comment:
Ah, yes, the comment I made of making host_id/email/source a key would be wrong anyway, because you can have multiple chrome profiles for the same host... more wins for not going that direction :D

Reply:
That would still work with all three being part of the pk.

Reply:
Oh right... man, where is my mind